proxy/benches/record.rs

```diff
@@ -48,6 +48,7 @@ where
         &proxy,
         &addr(),
         destination::Metadata::new(metrics::DstLabels::new(labels), None),
+        false
```

Comment:
What do you think about using either a newtype or an enum to more clearly illustrate what's going on?  `IsTls(false)` or `Tls::Disabled` are much easier to grok than `false`, all other things equal.

Reply:
I'm actually currently working on another branch based on this one that will change this to a `TlsStatus` enum, but I could backport that change to this branch if you like?

Reply:
If this is just going to change immediately, I'd prefer to review the targeted approach. Whether that's best backported to this branch or otherwise, I leave up to you.

Reply:
Hmm...the additional changes depend on #1148 as well, so I'll just backport the enum to this branch so you can review it here.